pkg/providers/tinkerbell/hardware/config.go

```diff
@@ -80,6 +81,10 @@ func (hc *HardwareConfig) ValidateHardware() error {
 			return fmt.Errorf("hardware: %s ID is required", hw.Name)
 		}
 
+		if _, ok := hardwares[hw.Spec.ID]; !ok {
+			return fmt.Errorf("hardware with the ID: %s is not present on the tinkerbell stack", hw.Spec.ID)
```

Comment:
The error seems a little confusing. `fmt.Errorf("found hardware uuid not registered with the tinkerbell stack: %v", id)`?

Reply:
IMHO, starting with "found hardware uuid ... " seems more confusing to me. We could stick with the original or keep it simple and say "Hardware ID %s not registered with tinkerbell... "

Reply:
Sure. It was less about the words and more about `hardware with the ID:` not quite making sense. `hardware id '%v' is not registered with tinkerbell` sounds good.